nexus/db-queries/src/db/datastore/rack.rs

```diff
@@ -629,6 +641,46 @@ impl DataStore {
                         })?;
                     }
 
+                    // Insert the RSS-generated blueprint.
+                    Self::blueprint_insert_on_connection(
+                        &conn,
+                        opctx,
+                        &blueprint,
+                        ).await
+                        .map_err(|e| {
+                            warn!(
+                                log,
+                                "Initializing Rack: Failed to insert blueprint"
+                            );
+                            err.set(RackInitError::BlueprintInsert(e)).unwrap();
+                            DieselError::RollbackTransaction
+                        })?;
+
+                    // Mark the RSS-generated blueprint as the current target,
+                    // DISABLED. We may change this to enabled in the future
+                    // when more of Reconfigurator is automated, but for now we
+                    // require a support operation to enable it.
+                    Self::blueprint_target_set_current_on_connection(
+                        &conn,
+                        opctx,
+                        BlueprintTarget {
+                            target_id: blueprint.id,
+                            enabled: false,
+                            time_made_target: Utc::now(),
+                        },
```

Comment:
We wouldn't really expect anything to be able to concurrently set a *different* target while we're doing this operation, would we? As in, other than this pathway, is "setting the target" still an operator/developer-driven operation?

Why I ask: I'm just curious if some other background task in Nexus could attempt to concurrently set a blueprint and cause this `blueprint_target_set` call to fail